jetbrains.*: move pkgs to by-name (#551369)
NixOS/nixpkgs: doc/release-notes/rl-2611.section.md

  Additionally, the kernel module tree used inside the VM has been split out of the `kernel` argument into a new `kernelModules` argument (defaulting to `kernel`).
  Callers that overrode `kernel` with a module tree (e.g. from `pkgs.aggregateModules`) to make extra modules available must now pass it via `kernelModules` instead, keeping `kernel` pointing at a bootable kernel derivation.

- The JetBrains IDEs have all been moved to the top-level:
  - `jetbrains.idea` and `jetbrains.idea-oss` have been renamed to `intellij-idea` and `intellij-idea-oss` respectively.

>   - MPS and Gateway require a `jetbrains-` prefix (e.g. `jetbrains.mps` -> `jetbrains-mps`).

Reviewer: Could match tha above explicit wording:

```suggestion
  - `jetbrains.mps` and `jetbrains.gateway` have been renamed to `jetbrains-mps` and `jetbrains-gateway` respectively.
```

Maybe even clearer if split into two bullets:
```suggestion
  - `jetbrains.mps` has been renamed to `jetbrains-mps`.
  - `jetbrains.gateway` has been renamed to `jetbrains-gateway`.
```

>   - All other IDEs have directly moved to the top-level with no prefix  (e.g. `jetbrains.pycharm` -> `pycharm`).

Reviewer: Maybe worth mentioning what hasn't moved yet? E.g.,

```markdown
  - `jetbrains.plugins` remains unchanged for now.
```

Author: Hm, maybe. But I doubt we will move `jetbrains.plugins` out, I don't know where it would even really make sense to put it, it's not really a package.

Reviewer: #### Release note feedback
The main thing is to further emphasise that only IDE packages have moved, not plugin packages. Someone skim-reading might assume that everything in the `jetbrains` package set has been moved to top-level attrs.

We already say "All other IDEs" instead of "All other attrs", but I suspect that's too subtle without further clarification.

#### Package set feedback
We also dropped all plugin derivations recently, so now `jetbrains.plugins` only contains `addPlugins` and `tests`.
- Long term, I'd like to see `jetbrains.plugins.addPlugins` replaced by a `<ide>.withPlugins` passthru function which wraps `finalAttrs.finalPackage` with plugins.
- `jetbrains.plugins.tests` could also be moved into the toplevel `tests` package-set, e.g. `tests.jetbrainsPlugins`?
- This would leave `jetbrains.plugins` empty.

Long term, it also begs the question of whether we should still have a `jetbrains` package set if all of its packages were moved to the top-level; where _should_ builders like `mkJetbrainsProduct` and plugins live? They're probably fine at the top-level, like most other builders:

<details><summary>Details</summary>
<p>

```
nix-repl> mk
mkBinaryCache         mkLLVMPackages        mkRocqPackages        mkShell               mkYarnModules         mkbootimg-osm0sis     mkchromecast          mkdocs                mkgmap-splitter       mklittlefs            mkp224o               mksh                  mktoc                 mkvtoolnix
mkCoqPackages         mkNginxPlugin         mkRuby                mkShellNoCC           mkYarnPackage         mkbrr                 mkcl                  mkfontdir             mkinitcpio-nfs-utils  mkmtkhdr              mkpasswd              mkspiffs              mktorrent             mkvtoolnix-cli
mkGCCNGPackages       mkNugetDeps           mkRubyVersion         mkStdenvNoLibs        mkalias               mkcal                 mkclean               mkfontscale           mkjson                mkosi                 mkrand                mkspiffs-presets      mktxp
mkKops                mkNugetSource         mkSaneConfig          mkTerraform           mkbootimage           mkcert                mkcue                 mkgmap                mkl                   mkosi-full            mkroot-utils          mktemp                mkuimage

nix-repl> make
make-minimal-bootstrap-sources  makeDarwinBundle                makeGCOVReport                  makeInitrdNG                    makeOverridable                 makeScopeWithSplicing'          makeStaticBinaries              makebootfat                     makejinja                       makeself
makeAutostartItem               makeDesktopItem                 makeHardcodeGsettingsPatch      makeInitrdNGTool                makePkgconfigItem               makeSetupHook                   makeStaticDarwin                makedepend                      makemkv
makeBinaryWrapper               makeFontsCache                  makeImpureTest                  makeLinuxHeaders                makeRustPlatform                makeShellWrapper                makeStaticLibraries             makefile2graph                  maker-panel
makeDBusConf                    makeFontsConf                   makeInitrd                      makeModulesClosure              makeScopeWithSplicing           makeStatic                      makeWrapper                     makehuman                       makerpm

nix-repl> build
build2                         buildDartApplication           buildEnv                       buildGo126Module               buildLakePackage               buildNimSbom                   buildRubyGem                   buildVscode                    buildbot-full                  buildcatrust                   buildkite-cli
build2-bootstrap               buildDotnetGlobalTool          buildFHSEnv                    buildGo127Module               buildLinux                     buildNpmPackage                buildRustCrate                 buildWasmBindgenCli            buildbot-plugins               buildifier                     buildkite-test-collector-rust
buildArmTrustedFirmware        buildDotnetModule              buildFHSEnvBubblewrap          buildGoLatestModule            buildMaven                     buildOptee                     buildRustCrateHelpers          buildXenPackage                buildbot-ui                    buildkit                       buildozer
buildBarebox                   buildDotnetPackage             buildFHSEnvChroot              buildGoModule                  buildMozillaMach               buildPackages                  buildTeleport                  buildah                        buildbot-worker                buildkit-nix                   buildpack
buildBazelPackage              buildDubPackage                buildGo123Module               buildGraalvmNativeImage        buildNavidromePlugin           buildPgrxExtension             buildTypstPackage              buildah-unwrapped              buildbotPackages               buildkite-agent                buildstream
buildBowerComponents           buildEmscriptenPackage         buildGo125Module               buildHomeAssistantComponent    buildNimPackage                buildPlatform                  buildUBoot                     buildbot                       buildbox                       buildkite-agent-metrics        buildtorrent
```

</p>
</details>
